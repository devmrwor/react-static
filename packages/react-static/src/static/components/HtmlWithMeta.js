import React from 'react'

// Instead of using the default components, we need to hard code meta
// from react-helmet into the components
export const makeHtmlWithMeta = ({ head }) => ({ children, ...rest }) => (
  <html lang="en" {...head.htmlProps} {...rest}>
    {children}
  </html>
)
