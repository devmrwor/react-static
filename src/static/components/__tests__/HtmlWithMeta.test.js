import React from 'react'
import { mount } from 'enzyme'
import { makeHtmlWithMeta } from '../HtmlWithMeta'

describe('HtmlWithMeta', () => {
  test('when route is a static route', () => {
    const HtmlWithMeta = makeHtmlWithMeta({
      head: { htmlProps: { lang: 'en' } },
    })

    const htmlWithMeta = mount(
      <HtmlWithMeta className="body">
        <head>
          <title>React Static</title>
        </head>
      </HtmlWithMeta>
    )

    expect(htmlWithMeta).toMatchSnapshot()
  })
})
