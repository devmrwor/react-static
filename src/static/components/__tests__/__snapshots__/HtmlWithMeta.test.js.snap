// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlWithMeta when route is a static route 1`] = `
<Component
  className="body"
>
  <html
    className="body"
    lang="en"
  >
    <head>
      <title>
        React Static
      </title>
    </head>
  </html>
</Component>
`
